### PR TITLE
Deny clippy warnings in CI and fix issues

### PR DESCRIPTION
**Pull request by JohnnyMorganz**
Denies warnings in CI for clippy, so that the workflow fails.
Fixes currently outstanding warnings

Note, I had to allow `unnecessary_wraps` in the `tokenizer.rs` file due to all the `advance_XXXX()` methods, as clippy currently triggers a false positive for these cases (https://github.com/rust-lang/rust-clippy/issues/6613, and maybe https://github.com/rust-lang/rust-clippy/issues/6517?)

### CODE REVIEW

File: full-moon/src/tokenizer.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::unnecessary_wraps)] // Clippy currently triggers false positives wrt. Advancement: https://github.com/rust-lang/rust-clippy/issues/6613
```
Comment (Kampfkarren) [inline full-moon/src/tokenizer.rs:1]: Where was this popping up? Only in a few places, or everywhere? If it's not in a lot of places, I'd rather filter those places specifically.

Comment (JohnnyMorganz) [reply full-moon/src/tokenizer.rs:1]: It was for all the following:
- `advance_identifier`
- `advance_shebang`
- `advance_symbol`
- `advance_whitespace`
The only one it was fine for was `advance_quote`

Comment (JohnnyMorganz) [reply full-moon/src/tokenizer.rs:1]: Following the merge of #154, this is no longer necessary. It has now been removed

**Comment by Kampfkarren** [comment]
Thanks!

**Pull request merged**